mvlearn/decomposition/baseica.py

```diff
@@ -0,0 +1,233 @@
+        to recover original data from reduced data.
+
+    multiview_output : bool, default True
+        If True, the `.transform` method returns one dataset per view.
+        Otherwise, it returns one dataset, of shape (n_samples, n_components)
+
+    random_state : int, RandomState instance or None, default=None
+        Used to perform a random initialization. If int, random_state is
+        the seed used by the random number generator; If RandomState
+        instance, random_state is the random number generator; If
+        None, the random number generator is the RandomState instance
+        used by np.random.
+
+    verbose : bool, default=False
+        Print information
+
+    Attributes
+    ----------
+    preproc_instance : ViewTransformer-like instance
+        The fitted instance used for preprocessing
+
+    mixing_ : array, shape (n_views, n_components, n_components)
+        The square mixing matrices, linking preprocessed data
+        and the independent components.
+
+    pca_components_: array, shape (n_components, n_features)
+        Principal axes in feature space, representing the directions
+        of maximum variance in the data. Only used if preproc == "pca".
+
+    components_ : array, shape (n_views, n_components, n_components)
+        The square unmixing matrices
+
+    individual_components_ : list of array
+        Individual unmixing matrices estimated by least squares.
+        `individual_components_[i]` is an array of shape
+        (n_components, n_features) where n_features is the number of
+        features in the dataset `i`.
+
+    individual_mixing_ : list of array
+        Individual mixing matrices estimated by least squares.
+        `individual_components_[i]` is an array of shape
+        (n_features, n_components) where n_features is the number of
+        features in the dataset `i`.
+    """
+
+    def __init__(
+        self,
+        n_components=None,
+        preproc="pca",
+        random_state=0,
+        verbose=False,
+        multiview_output=True,
+    ):
+        self.verbose = verbose
+        self.n_components = n_components
+        self.preproc = preproc
+        if preproc is None or preproc == "pca" and n_components is None:
+            self.preproc_instance = None
+            self.preproc_name = "None"
+        elif preproc == "pca":
+            self.preproc_name = "pca"
+            self.preproc_instance = ViewTransformer(
+                PCA(n_components=n_components)
+            )
+        else:
+            if hasattr(preproc, "transform") and hasattr(
+                preproc, "inverse_transform"
+            ):
+                self.preproc_name = "custom"
+                self.preproc_instance = preproc
+            else:
+                raise ValueError(
+                    "The dimension reduction instance needs"
+                    "a transform and inverse transform method"
+                )
```

Comment:
I don't like this preproc_name and preproc_instance thing.

it we need more attributes than preproc than they should appear in the fit and not the init. They must also end then with underscore

Reply:
Ok I'll remove the preproc_name we don't really need it